### DISCUSSION TITLE
String interpolation in queries.

### DISCUSSION THREAD

**Discussion by pfrang** [graphql/graphiql#3564] [Q&A / Support]
Hi, is it even possible to implement subqueries / partially valid queries that can be imported as const to another query while keeping the LSP support?
I notice that on importing substring consts Im losing some LSP support like formatting and  the file turns red and indicate error in VSCode, although the query works just fine.

Is there a workaround? If not, is there a possibility to set an ignore line, like a // @ts-ignore for the line "interfering" ?

<img width="611" alt="image" src="https://github.com/graphql/graphiql/assets/85552255/d9e020db-cb0f-4757-9679-591881625d3a">
<img width="286" alt="image" src="https://github.com/graphql/graphiql/assets/85552255/2b870eed-b371-41b5-8d3c-4413034e40a2">

**Comment by acao** [comment]
yes it's possible, but the expression is replaced with a newline in our LSP and other graphql LSPs. and an empty leaf expression is invalid in graphql. so there just needs to be at least one literal field to make the validation work. we and other graphql LSPs do not support anything beyond this, because we cannot infer the values. yes it's annoying but the only way around this is to disable the relevant graphql validation rules by providing a custom ruleset in graphql config

**Reply by pfrang** [reply]
Thanks for the reply! I notice that when a stirng interplation is included in the string, other errors are not longer displayed. Is it possible to define in the VSCode extension it should simply ignore everything wrapped in ${} for linting/formatting ?

**Reply by acao** [reply]
yes that's what we do, we replace them with whitespace (if we ignore them then all the offsets and jump to definitions are off), however the official graphql parser we use for validation treats empty leafs as syntax errors. so if you have a single template tag expression as the only field in a leaf, you will get a syntax error

also yes I noticed that validation had issues when there are multiple errors so there is a fix for that and many other issues coming that I've been working on the last month

**Reply by pfrang** [reply]
Ok, sounds great! Another issue, which I have seen other threads about but not specifically for this package, is the usage of the same fragment twice in a query which generates error. 
Is there some config that can be set up for this or is this expected to be handled in the gql func?
